staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1332,7 +1332,7 @@ func (v *unstructuredSchemaCoercer) apply(u *unstructured.Unstructured) (unknown
 			if v.returnUnknownFieldPaths {
 				pruneOpts.ReturnPruned = true
 			}
-			unknownFieldPaths = structuralpruning.PruneWithOptions(u.Object, v.structuralSchemas[gv.Version], false, pruneOpts)
```

Comment:
It seems that this PR changes the product code while description says it adds tests. Please clarify - update description or add a note why this change is needed